### PR TITLE
uint64_t support in emberAfFillExternalBuffer

### PR DESCRIPTION
**Pull request by shana-apple**
In the fabric cluster, I need to return a uint64_t fabricID so need emberAfFillExternalBuffer to support it

### CODE REVIEW

File: src/app/util/client-api.cpp [hunk 1]
```diff
@@ -181,12 +181,16 @@ static uint16_t vFillBuffer(uint8_t * buffer, uint16_t bufferLen, uint8_t frameC
             {
                 valueLen = 4;
             }
+            else if (cmd == 'y')
```
Comment (msandstedt) [inline src/app/util/client-api.cpp:184]: Can you update the comment in line 163 to 166 to reflect this addition?  Thanks.

File: src/app/util/client-api.cpp [hunk 2]
```diff
@@ -181,12 +181,16 @@ static uint16_t vFillBuffer(uint8_t * buffer, uint16_t bufferLen, uint8_t frameC
...
+            {
+                valueLen = 8;
+            }
             else
             {
                 emberAfDebugPrintln("ERR: Unknown format '%c'", cmd);
                 return 0;
             }
-            value = valueLen <= 2 ? static_cast<uint32_t>(va_arg(argPointer, int)) : va_arg(argPointer, uint32_t);
+            value = valueLen <= 2 ? static_cast<uint64_t>(va_arg(argPointer, int)) : va_arg(argPointer, uint64_t);
```
Comment (msandstedt) [inline src/app/util/client-api.cpp:193]: This isn't correct.  Making this change does not force the caller to pass 64-bit numbers for x,w,y.  So for any x and w commands that were correctly passing before, they will now have stack garbage in the lower or upper 4 bytes of `value`.

Also be advised that for the 'y' case, the caller must actually pass 64-bit types.  So importantly, passed literals need to be appended with 'ULL'.

Comment (msandstedt) [reply src/app/util/client-api.cpp:193]: Sorry, I should also clarify that this is specifically broken for platforms where integers are <= 32bits wide.

As a small aside, prior to these changes, the code was already broken for big-endian platforms with 64-bit integers.  That's probably not something we're likely to encounter.  But it's worth mentioning that this wasn't coming from a place of perfection before either.

TLDR: the rules of va_arg need to be honored.  This code does not.

File: src/app/util/client-api.cpp (hunk 1 above)
Comment (shana-apple) [reply src/app/util/client-api.cpp:184]: done

File: src/app/util/client-api.cpp (hunk 2 above)
Comment (shana-apple) [reply src/app/util/client-api.cpp]: Revised the code

**Review by msandstedt — APPROVED** [review]

**Review by woody-apple — APPROVED** [review]

File: src/app/util/client-api.cpp [hunk 3]
```diff
@@ -181,12 +181,27 @@ static uint16_t vFillBuffer(uint8_t * buffer, uint16_t bufferLen, uint8_t frameC
...
+            {
+                valueLen = 8;
+            }
             else
             {
                 emberAfDebugPrintln("ERR: Unknown format '%c'", cmd);
                 return 0;
             }
-            value = valueLen <= 2 ? static_cast<uint32_t>(va_arg(argPointer, int)) : va_arg(argPointer, uint32_t);
+            if (value <= 2)
```
Comment (bzbarsky-apple) [inline src/app/util/client-api.cpp:193]:
```suggestion
            if (valueLen <= 2)
```

File: src/app/util/client-api.cpp [hunk 4]
```diff
@@ -181,12 +181,27 @@ static uint16_t vFillBuffer(uint8_t * buffer, uint16_t bufferLen, uint8_t frameC
...
+            {
+                value = static_cast<uint64_t>(va_arg(argPointer, int));
+            }
+            else if (value <= 4)
```
Comment (bzbarsky-apple) [inline src/app/util/client-api.cpp:197]:
```suggestion
            else if (valueLen <= 4)
```

**Review by bzbarsky-apple — DISMISSED** [review]

**Review by msandstedt — APPROVED** [review]

**Review by Damian-Nordic — APPROVED** [review]
